### DISCUSSION TITLE
Using CMake on MacOS

### DISCUSSION THREAD

**Discussion by NiklasGustafsson** [xamarin/TorchSharp#1240] [Q&A]
I finally have a Mac with Apple Silicon, so I can start working on MPS support, and such. I'm having trouble getting CMake configured in the local build -- on Azure Pipelines, it's all set up and works, but it's not working for me on my local machine. It pops up a window asking for input and output locations, then tried to configure the Makefiles, but it's coming up with errors.

```
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in CMakeLists.txt:
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at LibTorchSharp/CMakeLists.txt:7 (find_package):
  Could not find a package configuration file provided by "Torch" with any of
  the following names:

    TorchConfig.cmake
    torch-config.cmake

  Add the installation prefix of "Torch" to CMAKE_PREFIX_PATH or set
  "Torch_DIR" to a directory containing one of the above files.  If "Torch"
  provides a separate development package or SDK, be sure it has been
  installed.


Configuring incomplete, errors occurred!
```

I installed CMake from the CMake org site. Maybe there's another version I should be using? It doesn't seem to be coming bundled with XCode.

**Comment by NiklasGustafsson** [comment]
@dayo05 -- you've been working on Mac, maybe you have some thoughts? I know it's been a while since you were actively doing it, but maybe you recall?

**Comment by dayo05** [comment]
> @dayo05 -- you've been working on Mac, maybe you have some thoughts? I know it's been a while since you were actively doing it, but maybe you recall?

@NiklasGustafsson The error shows that CMake cannot found dylib of LibTorch. Refer to my previous discussion(#837), Can you check did you set some environment variables to declare prefix directories where LibTorch located?

If this case, you can use command: `$ CMAKE_PREFIX_PATH=path/to/pytorch/torch/share/cmake Caffe2_DIR=path/to/pytorch/torch/share/cmake/Caffe2 cmake .` to generate makefile. This just set two temporal environment value about path of libtorch and caffe2 on makefile generation. 
Idk why I declare not only libtorch but also caffe2 but as I remember, I cannot compile without that environment value.

I've wrote that reference very long time ago and never updated so it is based on codebase in 2022 but it seems CMakeLists not contains any major changes from there.

**Reply by dayo05** [reply]
Because my disk was full so I cannot test with it right now but Ill plan to do it asap

**Reply by NiklasGustafsson** [reply]
I get the same error message window popping up, with the same error message.

**Reply by NiklasGustafsson** [reply]
Following your instructions, I can't even build pytorch on my machine, with similar cmake errors. There must be something wrong with my setup.

**Reply by NiklasGustafsson** [reply]
Installed 'cmake' with homebrew instead of the installer, and it's building now... :-)
